js/runner.go

```diff
@@ -385,7 +390,27 @@ type ActiveVU struct {
 // Activate the VU so it will be able to run code.
 func (u *VU) Activate(params *lib.VUActivationParams) lib.ActiveVU {
 	u.Runtime.ClearInterrupt()
-	// u.Env = params.Env
+
+	if params.Exec == "" {
+		params.Exec = consts.DefaultFn
+	}
+
+	// Override the preset global env with any custom env vars
+	if len(params.Env) > 0 {
+		env := u.Runtime.Get("__ENV").Export().(map[string]string)
+		for key, value := range params.Env {
+			env[key] = value
+		}
+		u.Runtime.Set("__ENV", env)
+	}
+
+	if len(params.Tags) > 0 {
+		tags := u.Runner.Bundle.Options.RunTags.CloneTags()
+		for k, v := range params.Tags {
+			tags[k] = v
+		}
+		u.Runner.Bundle.Options.RunTags = stats.IntoSampleTags(&tags)
```

Comment:
hmm this seems wrong - if I'm understanding it right, it means that one executor can "taint" the tags of the VU, so they persist even in follow-up executors?

Reply:
Yeah, this is wrong
```js
export let options = {
  execution: {
    scenario1: {
      type: "per-vu-iterations",
      vus: 1,
      iterations: 1,
      maxDuration: "3s",
      tags: {
        "111": "111",
      },
    },
    scenario2: {
      type: "per-vu-iterations",
      vus: 1,
      iterations: 1,
      startTime: "3s",
      maxDuration: "10s",
      tags: {
        "222": "222",
      },
    },
  },
};
export default function () {
  console.log(`Start VU ${__VU} ITER ${__ITER}`);
}
```

running that with `--out json` would produce metrics like this:
```
{"type":"Point","data":{"time":"2020-05-08T17:57:14.891533472+03:00","value":1,"tags":{"111":"111","group":""}},"metric":"iterations"}
{"type":"Point","data":{"time":"2020-05-08T17:57:17.891892686+03:00","value":1,"tags":{"111":"111","222":"222","group":""}},"metric":"iterations"}
```

Reply:
I remember there being code that reverted the per executor ones at some point :thinking:

Reply:
This should be fixed by ae5ef34f.

I considered moving `runFn` to `ActiveVU` instead of `VU`, since `VU` shouldn't be running anything, but that complicated things with `runPart`, and I didn't want to get into that refactor right now.

Instead, I'm just passing the custom tags to `runFn` for the `netext.NetTrail` tags, and for HTTP, WS, checks and group metric tags it was easiest to add a `Tags` field to `lib.State`, which seems like an elegant, albeit disjointed, solution.

Let me know.